ansible/lifecycle.yml

```diff
@@ -68,4 +75,74 @@
                   {% for instance in r_instances.instances %}
                   {{ "%-60s" | format(instance.tags.Name) }} {{ "%-10s" | format(instance.state.name) }} {{ instance.instance_type }}
                   {% endfor %}
+    - when: cloud_provider == 'osp'
+      environment:
+        OS_AUTH_URL: "{{ osp_auth_url }}"
+        OS_USERNAME: "{{ osp_auth_username }}"
+        OS_PASSWORD: "{{ osp_auth_password }}"
+        OS_PROJECT_NAME: "{{ osp_project_name }}"
+        OS_PROJECT_DOMAIN_ID: "{{ osp_auth_project_domain }}"
+        OS_USER_DOMAIN_NAME: "{{ osp_auth_user_domain }}"
+      block:
+        - when: ACTION == 'stop'
+          block:
+            - name: Gather instance facts
+              os_server_info:
+                filters:
+                  metadata:
+                    Project: "{{ project_tag }}"
+                    guid: "{{ guid }}"
+                    env_type: "{{ env_type }}"
+                  vm_state: active
+              register: r_osp_facts
+
+            - when: r_osp_facts.openstack_servers | length > 0
+              block:
+                - set_fact:
+                    all_instances: >-
+                      {{ r_osp_facts.openstack_servers
+                      | json_query('[*].id') }}
+
+                - name: Stop all servers
+                  command: nova stop {{ all_instances | join(' ') }}
+
+        - when: ACTION == 'start'
+          block:
+            - name: Gather instance facts
+              os_server_info:
+                filters:
+                  metadata:
+                    guid: "{{ guid }}"
+                    env_type: "{{ env_type }}"
+                  vm_state: stopped
+              register: r_osp_facts
+
+            - when: r_osp_facts.openstack_servers | length > 0
+              block:
+                - set_fact:
+                    all_instances: >-
+                      {{ r_osp_facts.openstack_servers
+                      | json_query('[*].id') }}
 
+                - name: Start all servers
+                  command: nova start {{ all_instances | join(' ') }}
```

Comment:
Please consider using the openstack client here instead of nova.

Reply:
Will do, thanks